aviasales/src/app/app.component.ts

```diff
@@ -0,0 +1,77 @@
+import { Component, OnInit } from '@angular/core';
+import { Ticket, TicketsService } from './services/tickets.service';
+
+export interface TicketsParams {
+  sort: 'cheapest' | 'fastest';
+  amount: number;
+  filters: number[];
+}
+
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.sass'],
+})
+export class AppComponent implements OnInit {
+  isCompleted = false;
+
+  tickets: Ticket[] = [];
+
+  filteredTickets: Ticket[] = [];
+
+  shownTickets: Ticket[];
+
+  ticketsParams: TicketsParams = {
+    sort: 'cheapest',
+    amount: 5,
+    filters: [0, 1, 2, 3],
+  };
+
+  constructor(private ticketsService: TicketsService) {}
+
+  ngOnInit(): void {
+    this.ticketsService.fetchTickets().then((observer) => {
+      observer.subscribe((ticketPack) => {
```

Comment:
you should unsubscribe in onDestroy hook

Reply:
if you'll use only **observable**, you'll need only **subscribe**, without **then**